pkg/remote/remote.go

```diff
@@ -14,20 +14,415 @@
 package remote
 
 import (
+	"context"
```

Comment:
This contains the logic of the former `deploy/remote.go` and `destroy/remote.go` which was duplicated for both scenarios. Now it is common in this place